src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -697,6 +697,9 @@ void MutationsInterpreter::prepare(bool dry_run)
                     type_literal);
 
                 stages.back().column_to_updated.emplace(column, updated_column);
+
+                if (condition && return_mutated_rows)
+                    stages.back().filters.push_back(condition);
```

Comment:
How `ALTER UPDATE WHERE` worked before if we didn't apply filters?

Reply:
It correctly mutates the rows it needs by creating `SELECT if(condition, mutate, old_value)...`. The problem for KV store is that MutationIntepreter returns all the rows causing an update on ALL keys while we only need a subset we would like to override.

Reply:
Thanks. So, for rocksdb we will not rewrite values with the same `old_value`. And for keepermap we will also not update `_version`.

Reply:
exactly 😄